semgrep-core/src/reporting/Report.ml

```diff
@@ -352,7 +353,12 @@ let make_final_result results rules ~rules_parse_time =
   let extra =
     let mk_profiling () =
       let file_times = results |> Common.map get_profiling in
-      { rules; rules_parse_time; file_times }
+      {
+        rules;
+        rules_parse_time;
+        file_times;
+        max_memory_bytes = (Gc.quick_stat ()).top_heap_words;
```

Comment:
This is a good start. Let's leave a comment talking about some of the caveats we discussed. Most importantly, if the most amount of memory is consumed after forking, the real max needs to be multiplied by the number of processes.